colab-notebooks-examples/integrations/openai/intro.ipynb

```diff
@@ -0,0 +1,408 @@
+    "        yield {\n",
+    "            \"_index\": 'wikipedia_vector_index',\n",
+    "            \"_id\": row['id'],\n",
+    "            \"_source\": {\n",
+    "                'url' : row[\"url\"],\n",
+    "                'title' : row[\"title\"],\n",
+    "                'text' : row[\"text\"],\n",
+    "                'title_vector' : json.loads(row[\"title_vector\"]),\n",
+    "                'content_vector' : json.loads(row[\"content_vector\"]),\n",
+    "                'vector_id' : row[\"vector_id\"]\n",
+    "            }\n",
+    "        }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8164b38",
+   "metadata": {},
+   "source": [
+    "As the dataframe is large, we will index data in batch of `100`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aacb5e9c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from elasticsearch import helpers\n",
+    "\n",
+    "start = 0\n",
+    "end = len(wikipedia_dataframe)\n",
+    "batch_size = 100\n",
+    "for batch_start in range(start, end, batch_size):\n",
+    "    batch_end = min(batch_start + batch_size, end)\n",
+    "    batch_dataframe = wikipedia_dataframe.iloc[batch_start:batch_end]\n",
+    "    actions = dataframe_to_bulk_actions(batch_dataframe)\n",
+    "    helpers.bulk(client, actions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "091ffc51",
+   "metadata": {},
+   "source": [
+    "Let's test the index with simple match query."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2ccc8955",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(client.search(index=\"wikipedia_vector_index\", query={\n",
+    "    \"match\": {\n",
+    "      \"text\": {\n",
+    "        \"query\": \"Hummingbird\"\n",
+    "      }\n",
+    "    }\n",
+    "}))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "992b6804",
+   "metadata": {},
+   "source": [
+    "# Encode a question with OpenAI embedding model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6cd1839c",
+   "metadata": {},
+   "source": [
+    "Set OpenAI API key and Create a new OpenAI embedding using  `text-embedding-ada-002` model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "57385c69",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai\n",
+    "OPENAI_API_KEY = getpass(\"Enter OpenAI API key\")\n",
+    "\n",
+    "openai.api_key = OPENAI_API_KEY"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ca4c51ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "EMBEDDING_MODEL = \"text-embedding-ada-002\"\n",
+    "\n",
+    "question = 'How wide is Atlantic ocean?'\n",
+    "modelOpenAi = openai.Embedding.create(input=question, model=EMBEDDING_MODEL)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7e6bf5d",
+   "metadata": {},
+   "source": [
+    "# Perform kNN search"
```

Comment:
Title should be more around "Performing Queries"

Reply:
plus also mention what you're doing here. Links to documentation for KNN search would be useful too.

Reply:
Should use `##` for all H2s too